In [1]:
import pandas as pd
import numpy as np

In [2]:
from google.colab import files

uploaded = files.upload()

Saving reporte_clinica.csv to reporte_clinica.csv


In [3]:
df = pd.read_csv("reporte_clinica.csv")

In [4]:
df.head()

,Unnamed: 0,age,sex,bmi,children,smoker,region,charges
0,0,19.0,female,27.900,0,yes,southwest,16884.92400
1,1,18.0,male,33.770,1,no,SE,1725.55230
2,2,28.0,male,33.000,3,no,southeast,4449.46200
3,3,33.0,male,22.705,0,no,northwest,21984.47061
4,4,32.0,male,28.880,0,no,northwest,3866.85520


In [5]:
df.shape

(1363, 8)

El dataset contiene alrededor de 1363 entradas, con ocho columnas adjacentes

In [6]:
list(df.columns)

['Unnamed: 0', 'age', 'sex', 'bmi', 'children', 'smoker', 'region', 'charges']

In [7]:
df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 1363 entries, 0 to 1362
Data columns (total 8 columns):
 #   Column      Non-Null Count  Dtype  
---  ------      --------------  -----  
 0   Unnamed: 0  1363 non-null   int64  
 1   age         1261 non-null   float64
 2   sex         1363 non-null   object 
 3   bmi         1313 non-null   float64
 4   children    1363 non-null   int64  
 5   smoker      1363 non-null   object 
 6   region      1363 non-null   object 
 7   charges     1363 non-null   float64
dtypes: float64(3), int64(2), object(3)
memory usage: 85.3+ KB


Se observan variables numéricas y categóricas. Algunas columnas corresponden a datos cuantitativos mientras que otras almacenan información textual.

In [8]:
df.describe()

,Unnamed: 0,age,bmi,children,charges
count,1363.00000,1261.000000,1313.000000,1363.000000,1363.000000
mean,681.00000,39.203807,34.758256,1.414527,13255.535475
std,393.60852,14.061447,65.717650,5.483174,12110.399149
min,0.00000,18.000000,-5.000000,-1.000000,1121.873900
25%,340.50000,26.000000,26.180000,0.000000,4703.000275
50%,681.00000,39.000000,30.400000,1.000000,9361.326800
75%,1021.50000,51.000000,34.675000,2.000000,16717.010750
max,1362.00000,64.000000,999.000000,99.000000,63770.428010


In [9]:
df.describe(include="object")

,sex,smoker,region
count,1363,1363,1363
unique,4,4,8
top,male,no,southeast
freq,653,1034,356


In [11]:
df.isnull().sum()

,0
Unnamed: 0,0
age,102
sex,0
bmi,50
children,0
smoker,0
region,0
charges,0


In [13]:
df.duplicated().sum()

np.int64(0)

In [14]:
for columna in df.columns:
    print(columna)
    print(df[columna].unique())
    print("-"*40)

Unnamed: 0
[   0    1    2 ... 1360 1361 1362]
----------------------------------------
age
[19. 18. 28. 33. 32. 31. 46. 37. 60. 25. 62. 23. 56. 27. 52. 30. 34. 59.
 63. 55. nan 26. 35. 24. 41. 38. 21. 48. 40. 58. 43. 64. 20. 61. 53. 44.
 57. 29. 22. 45. 54. 49. 47. 51. 42. 50. 36. 39.]
----------------------------------------
sex
['female' 'male' 'Female' 'Male']
----------------------------------------
bmi
[ 27.9    33.77   33.     22.705  28.88   25.74   33.44   27.74   29.83
  25.84   26.22   26.29   34.4    39.82   42.13   24.6    30.78   23.845
  40.3    35.3    36.005     nan  34.1    31.92   28.025  27.72   23.085
  32.775  17.385  36.3    35.6    26.315  28.6    28.31   20.425  32.965
  20.8    36.67   39.9    26.6    36.63   21.78   30.8    37.05   37.3
  38.665  34.77   24.53   35.2    35.625  33.63   28.     34.43   28.69
  36.955  31.825  31.68   22.88   37.335  27.36   33.66   24.7    25.935
  22.42   28.9    39.1    36.19   23.98   24.75   28.5    28.1    32.01
  27.4   

En esta corta revision se puede apreciar que, no solo hay datos faltantes si no tembien, hay, datos erroneos, como, por ejmplo, uno de los pacientes infantes teninedo una edad de -1